core/src/main/java/org/infinispan/health/impl/ClusterHealthImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.health.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.health.ClusterHealth;
+import org.infinispan.health.HealthStatus;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+public class ClusterHealthImpl implements ClusterHealth {
```

Comment:
Wrong name I think, this reports the health of a single manager, and not the health of the entire cluster.